Mathlib/CategoryTheory/Preadditive/AdditiveFunctor.lean

```diff
@@ -347,12 +348,14 @@ noncomputable def endRingEquiv
     (sq₀ : e.congrLeft.functor.obj f ≅ e'.congrRight.inverse.obj g) : End f ≃+* End g where
   __ := endMonoidEquiv sq₀
   map_add' α β := by
+    haveI : e.functor.Additive := e.functor.additive_of_preserves_binary_products
+    haveI : e'.functor.Additive := e'.functor.additive_of_preserves_binary_products
```

Comment:
It's nice that Lean already knows `additive_of_preserves_binary_products`; however I think it's better to keep the original `Additive` assumptions since they are more general. It turns out we only need that `e'` is additive, see [my branch](https://github.com/leanprover-community/mathlib4/commit/5d78a65d1cfbdbf1e53849feb395930e59ca58e1).
I've realized [my comment](https://github.com/leanprover-community/mathlib4/pull/20652/files#r1937614282) was misplaced and doesn't really belong to this PR.